SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp

```diff
@@ -186,19 +167,97 @@ QueryConstraint InterestFactory::CreateSystemDefinedConstraints()
+	TMap<UClass*, float> ActorCullDistances;
+	for (TObjectIterator<UClass> It; It; ++It)
+	{
+		if (!It->IsChildOf<AActor>())
+		{
+			continue;
+		}
 
-QueryConstraint InterestFactory::CreateCheckoutRadiusConstraint()
-{
-	QueryConstraint CheckoutRadiusConstraint;
+		const AActor* IteratedDefaultActor = Cast<AActor>(It->GetDefaultObject());
+		if (IteratedDefaultActor->NetCullDistanceSquared > DefaultDistanceSquared)
+		{
+			ActorCullDistances.Add(*It, IteratedDefaultActor->NetCullDistanceSquared);
+		}
+	}
+
+	// Sort the map for the next iteration so that base classes are seen before derived classes. This lets us skip
+	// derived classes that have a smaller cull distance than a parent class.
+	ActorCullDistances.KeySort([](UClass& LHS, UClass& RHS) {
+		return
+			LHS.IsChildOf(&RHS) ? -1 :
+			RHS.IsChildOf(&LHS) ? 1 :
+			0;
+	});
+
+	// If an actor's cull distance is smaller than that of a parent class, there's no need to add interest for that actor.
+	// Can't do inline removal since the sorted order is only guaranteed when the map isn't changed.
+	TMap<UClass*, float> OptimizedCullDistances;
+	for (const auto& ActorCullDistance : ActorCullDistances)
+	{
+		bool bShouldAdd = true;
+		for (auto& OptimizedCullDistance : OptimizedCullDistances)
+		{
+			if (ActorCullDistance.Key->IsChildOf(OptimizedCullDistance.Key) && ActorCullDistance.Value <= OptimizedCullDistance.Value)
+			{
+				// No need to add this cull distance since it's captured in the optimized map already.
+				bShouldAdd = false;
+				break;
+			}
+		}
+		if (bShouldAdd)
+		{
+			OptimizedCullDistances.Add(ActorCullDistance.Key, ActorCullDistance.Value);
+		}
+	}
+
+	QueryConstraint CheckoutRadiusConstraints;
+
+	// Use AActor's NetCullDistance for the default radius (all actors in that radius will be checked out)
+	const float DefaultCheckoutRadius= FMath::Sqrt(DefaultDistanceSquared / (100.0f * 100.0f));
+	QueryConstraint DefaultCheckoutRadiusConstraint;
+	DefaultCheckoutRadiusConstraint.RelativeCylinderConstraint = RelativeCylinderConstraint{ DefaultCheckoutRadius };
+	CheckoutRadiusConstraints.OrConstraint.Add(DefaultCheckoutRadiusConstraint);
+
+	// For every cull distance that we still want, add a constraint with the distance for the actor type and all of its derived types.
+	for (const auto& CullDistanceSquared : OptimizedCullDistances)
+	{
+		QueryConstraint CheckoutRadiusConstraint;
+
+		QueryConstraint RadiusConstraint;
+		const float CheckoutRadius = FMath::Sqrt(CullDistanceSquared.Value / (100.0f * 100.0f));
+		RadiusConstraint.RelativeCylinderConstraint = RelativeCylinderConstraint{ CheckoutRadius };
+		CheckoutRadiusConstraint.AndConstraint.Add(RadiusConstraint);
 
-	float CheckoutRadius = Actor->CheckoutRadius / 100.0f; // Convert to meters
-	CheckoutRadiusConstraint.RelativeCylinderConstraint = RelativeCylinderConstraint{ CheckoutRadius };
+		QueryConstraint ActorTypeConstraint;
```

Comment:
Unreal coding standard: encouraged to prefix variable name with 'Out' in the case that it is pass by reference into a function that is expected to write to it

Reply:
I see you've done that on the function itself.  I typically also do it on the actual variable I pass into the function.  Curious your thoughts on this one.